waku/v2/protocol/waku_swap/waku_swap.nim

```diff
@@ -20,6 +20,7 @@
 ##
 ## Things like settlement is for future work.
 ##
+# {.push raises: [Defect].}
```

Comment:
```suggestion

# {.push raises: [Defect].}
```

Could you add a brief comment on the immediate error you noticed here?

Reply:
Sure, done!

Reply:
I explained the initial error that I noticed, once I tried to resolve it by throwing the exception from the procs, further issues appeared that made me decide to deal with them in a separate PR

Reply:
Yeah, sympathy - you start with a simple `raises` and end up having to go down a rabbit hole in referenced modules.

Reply:
Thanks! ;) :D